include/clang/AST/Expr.h

```diff
@@ -4675,6 +4680,57 @@ class RangeBoundsExpr : public BoundsExpr {
   }
 };
 
+/// \brief Represents a Checked C interop bounds annotation.
+///
+/// Checked C has bounds-safe interfaces that allow global variables,
+/// functions, and members defined by unchecked code to be used in ch+ecked
+/// contexts and treated as having checked pointer types. This annotation is
+/// used to declare that a variable, parameter, or member has a bounds-safe
+//  interface that is a checked pointer type.
+///
+/// Typically this is used to declare an unchecked-pointer typed entity
+/// as having a _Ptr type.  There is no bounds expression for this.  More
+/// generally, it can declare a bounds-safe interface this is a checked
+/// pointer to a checked pointer.  This is useful for unchecked declarations
+///  such as `int **y', where y could have a bounds-safe interface that
+/// is `ptr<ptr<int>>` or `array_ptr<ptr<int>>`.
+///
+/// We typically need bounds-safe interface information at the same points
+/// where we need bounds information, so it is convenient to store the
+/// information as a bounds expression.
+class InteropTypeBoundsAnnot : public BoundsExpr {
```

Comment:
clang has an 80 character line limit, so long identifier names can cause problems with fitting expressions or parameter lists on to the line.   I agree that a more descriptive name is better, so I'll update the name.